src/now.js

```diff
@@ -437,10 +437,22 @@ const main = async (argv_) => {
       ctx.authConfig.credentials[credentialsIndex] = obj
     }
 
-    const user = await getUser({
-      apiUrl: ctx.apiUrl,
-      token
-    })
+    let user
+
+    try {
+      user = await getUser({
+        apiUrl: ctx.apiUrl,
+        token
+      })
+    } catch (err) {
+      console.error(error(err))
+      await exit(1)
+    }
+
+    // Don't use team from config if `--token` was set
+    if (ctx.config.sh && ctx.config.sh.currentTeam) {
+      delete ctx.config.sh.currentTeam
```

Comment:
Not a fan of changing the source of truth. I think that should be handled by the consumer of this function.